proposals/0000-markdown-content-redesign.md

```diff
@@ -0,0 +1,126 @@
+- Start Date: 03-10-2022
+- Reference Issues: https://github.com/withastro/rfcs/discussions/5, https://github.com/withastro/rfcs/discussions/118
+- Implementation PR: <!-- leave empty -->
+
+# Summary
+
+A v2.0 API for working with local Markdown files in Astro.
+
+# Example
+
+```astro
+---
+// BASIC USAGE:
+
+// Note: Astro.fetchContent() removed in favor of direct `import.meta.globEager` usage
+// Note: This works in non-Astro framework components!
+const markdownFilesObj = import.meta.globEager('../posts/*.md');
```

Comment:
So this is true, but I always found it odd how `fetchContent` was restricted to markdown alone. `import.meta.globEager` is a pretty convenient way to see all available routes for a directory in a mix of `.astro` or `.md` formats. Serves as an [11ty collections alternative](https://www.11ty.dev/docs/collections/) in a way.

Because of that, I'd love to start pushing users towards this more generic API so they don't feel restricted. Maintaining both `fetchContent` and `import.meta.glob` feels a bit odd to me. As a user, I'd worry that `globEager` is less performant or optimized compared to `fetchContent` since anything starting with `Astro.` feels like the right way :tm:.

Reply:
Thanks for the feedback yall! Based on this convo I think I have a new approach that keeps the idea of a helper (so that users don't need to mess with `import.meta.glob/globEager` for basic use) but makes that helper for generally useful (supports non-MD files).

Still updating the RFC, but you can see the new API in the example.